infra/modules/service/database-access.tf

```diff
@@ -14,9 +14,24 @@ resource "aws_vpc_security_group_ingress_rule" "db_ingress_from_service" {
   referenced_security_group_id = aws_security_group.app.id
 }
 
+
 resource "aws_iam_role_policy_attachment" "app_db_access" {
   count = var.db_vars != null ? 1 : 0
 
   role       = aws_iam_role.service.name
   policy_arn = var.db_vars.access_policy_arn
 }
```

Comment:
when are we planning on removing this block?

optional idea: what if we attached the migrator access policy to the app_service role rather than to the migrator role in the `migrator_db_access` resource below? then we can remove this policy attachment.

in the next PR when we update run-command.sh and run-database-migrations.sh to override the migration task to use the migrator role, we can then update the migrator_db_access policy attachment to attach to the migrator role instead of the app role.

in either case, an optional idea to add a # TODO comment of what we need to do. if we stick with the current PR's approach add a "TODO: delete this app_db_access policy attachment once we've successfully updated migrations to use the migrator role", or if we go with the idea in this comment, we'd add TODO in the migrator_db_access resource that says "TODO: update role to be migrator role once we update run-database-migrations.sh to override the task role to be the migrator role"

Reply:
I was planning on removing it as part of step 3. So next PR. For this PR, the original role is still used for migration tasks
https://github.com/navapbc/template-infra/issues/354#issuecomment-1693973424

Reply:
i think you'll find that that won't work. because in this PR the task definition is set to the app_service role, but the app_service role only has the app_access_policy. that means when we call run-database-migrations.sh, the migration script will try to log in as the migrator user but since the task role only has the app_service policy it won't be able to.

Reply:
Added todo comments